### PR TITLE
SSL vs. TLS consistency and clarification in documentation

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

With TLS landing in the stable v1.16.0 release I wanted to make sure that we're being clear and as consistent as possible about the fact that we're using TLS 1.2 as the encryption protocol, not any of the obsolete SSL protocols.

The fact that "SSL" is colloquially used to refer to any encrypted web connection—it's kind of like "Kleenex" being used for any brand of tissue—complicates this a bit. In certain places I try to ease the transition with `TLS/SSL`, but in my opinion we should be consistent in using `TLS` whenever possible, as that is technically correct.

##### Component Name

docs

##### Additional Information

I've also included some grammatical/stylistic fixes to the areas relevant to TLS.

In an ideal world, any references in configuration files are changed from `ssl` to `tls`, but that's probably a huge project without any real value for the user. For example:

```
[stream]
    destination = host:port:SSL #old
    destination = host:port:TLS #new
```

### CODE REVIEW

File: docs/Running-behind-nginx.md [hunk 1]
```diff
@@ -133,15 +132,15 @@ If nginx is not configured as described here, you will probably receive the erro
...
+Create an authentication file to enable basic authentication via Nginx. Do not use authentication without having first [enabled TLS](#using-tls-communication)!
+
+If you don't have an authentication file, you can use the following command:
 
 ```
 printf "yourusername:$(openssl passwd -apr1)" > /etc/nginx/passwords
 ```
 
-And enable the authentication inside your server directive:
+And thenenable the authentication inside your server directive:
```
Comment (vlvkobal) [inline docs/Running-behind-nginx.md]: A missed whitespace `then enable`

Comment (joelhans) [reply docs/Running-behind-nginx.md]: Got it. Thanks!

**Comment by cakrit** [comment]
I'll approve when @vlvkobal and @thiagoftsm approve.

**Comment by joelhans** [comment]
@thiagoftsm: Fixed that sentence you requested!

**Pull request merged**